app/src/docs/v1.api-spec.yaml

```diff
@@ -0,0 +1,2744 @@
+          description: Note updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DB-Note'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        "422":
+          $ref: "#/components/responses/UnprocessableEntity"
+        default:
+          $ref: '#/components/responses/Error'
+    delete:
+      summary: Delete a note
+      description: Soft deletes a note by marking isDeleted true
+      operationId: deleteNote
+      tags:
+        - Note
+      parameters:
+        - $ref: '#/components/parameters/Path-NoteId'
+      responses:
+        '200':
+          description: Note deleted successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DB-Note'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          $ref: '#/components/responses/Error'
+  /note/bringForward:
+    get:
+      summary: List notes with bring forward type
+      description: Retrieve a list of notes with the Bring forward type given
+      operationId: listBringForward
+      tags:
+        - Note
+      parameters:
+        - $ref: '#/components/parameters/Query-BringForwardState'
+      responses:
+        '200':
+          description: A list of bring forward notes
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Response-BringForwardNotes'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        default:
+          $ref: '#/components/responses/Error'
+  /note/list/{activityId}:
+    get:
+      summary: List notes by Activity Id
+      description: Return a list of notes that have the given activityId
+      operationId: listNotes
+      tags:
+        - Note
+      parameters:
+        - $ref: '#/components/parameters/Path-ActivityId'
+      responses:
+        '200':
+          description: A list of notes for the specified activity
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/DB-Note'
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        "422":
+          $ref: "#/components/responses/UnprocessableEntity"
+        default:
+          $ref: '#/components/responses/Error'
+  /permit/:
+    get:
+      summary: List permits by Activity ID
```

Comment:
For consistency, we should decide if we are fully capitalizing `ID`, or using `Id`.